go/serialization.go

```diff
@@ -350,31 +350,59 @@ func SerializeSCTSignatureInput(sct SignedCertificateTimestamp, entry LogEntry)
 	}
 }
 
-func serializeV1SCT(sct SignedCertificateTimestamp) ([]byte, error) {
-	if err := checkExtensionsFormat(sct.Extensions); err != nil {
-		return nil, err
-	}
-	var buf bytes.Buffer
-	if err := binary.Write(&buf, binary.BigEndian, V1); err != nil {
-		return nil, err
-	}
-	if err := binary.Write(&buf, binary.BigEndian, sct.LogID); err != nil {
-		return nil, err
+func serializeV1SCTHere(sct SignedCertificateTimestamp, here []byte) (int, error) {
+	if sct.SCTVersion != 0 {
+		return 0, INVALID_VERSION
 	}
-	if err := binary.Write(&buf, binary.BigEndian, sct.Timestamp); err != nil {
-		return nil, err
+
+	//Determine Lenth of Serialized Output
+	ctxlen := len(sct.Extensions)
+	siglen := len(sct.Signature.Signature)
+	sctoutput := 1 + 32 + 8 + 2 + ctxlen + 2 + 2 + siglen
+
+	if sctoutput > 0xFFFF {
```

Comment:
Could you add a comment with a reference to the 6962 section which defines this size limit?  Better still: add the `0xFFFF` as an exported constant with that comment at the top of this file.